framework/src/vectorpostprocessors/SpatialUserObjectVectorPostprocessor.C

```diff
@@ -47,24 +50,35 @@ SpatialUserObjectVectorPostprocessor::SpatialUserObjectVectorPostprocessor(
 void
 SpatialUserObjectVectorPostprocessor::fillPoints()
 {
-  if (isParamValid("points") && isParamValid("points_file"))
-    mooseError(name(), ": Both 'points' and 'points_file' cannot be specified simultaneously.");
-
-  if (isParamValid("points"))
+  if (_use_points_from_uo)
   {
-    _points = getParam<std::vector<Point>>("points");
+    if (isParamValid("points") || isParamValid("points_file"))
+      mooseWarning("When 'use_points_from_uo' is true, the 'points' and 'points_file' parameters "
+                   "are ignored");
+
+    _points = _uo.spatialPoints();
   }
-  else if (isParamValid("points_file"))
+  else
   {
-    const FileName & points_file = getParam<FileName>("points_file");
+    if (isParamValid("points") && isParamValid("points_file"))
+      mooseError(name(), ": Both 'points' and 'points_file' cannot be specified simultaneously.");
 
-    MooseUtils::DelimitedFileReader file(points_file, &_communicator);
-    file.setFormatFlag(MooseUtils::DelimitedFileReader::FormatFlag::ROWS);
-    file.read();
-    _points = file.getDataAsPoints();
+    if (isParamValid("points"))
+    {
+      _points = getParam<std::vector<Point>>("points");
+    }
+    else if (isParamValid("points_file"))
+    {
+      const FileName & points_file = getParam<FileName>("points_file");
+
+      MooseUtils::DelimitedFileReader file(points_file, &_communicator);
+      file.setFormatFlag(MooseUtils::DelimitedFileReader::FormatFlag::ROWS);
+      file.read();
+      _points = file.getDataAsPoints();
+    }
+    else
+      mooseError(name(), ": You need to supply either 'points' or 'points_file' parameter.");
```

Comment:
Don't need to use name(), add exception test

Reply:
I can add a test if it doesn't exist already - the diffs in this PR are just changes in indentation of already-existing code. But happy to add one :)

Reply:
Test has been added!